src/charm.py

```diff
@@ -1,117 +1,296 @@
-            if container.can_connect():
-                # Push an updated layer with the new config
-                container.add_layer("httpbin", self._pebble_layer, combine=True)
-                container.replan()
-
-                logger.debug("Log level for gunicorn changed to '%s'", log_level)
-                self.unit.status = ops.ActiveStatus()
-            else:
-                # We were unable to connect to the Pebble API, so we defer this event
-                event.defer()
-                self.unit.status = ops.WaitingStatus("waiting for Pebble API")
-        else:
-            # In this case, the config option is bad, so block the charm and notify the operator.
-            self.unit.status = ops.BlockedStatus("invalid log level: '{log_level}'")
-
-    @property
-    def _pebble_layer(self) -> pebble.LayerDict:
-        """Return a dictionary representing a Pebble layer."""
-        return {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {
-                        "GUNICORN_CMD_ARGS": f"--log-level {self.model.config['log-level']}"
-                    },
+        hostname = event.params["hostname"]
+        tls_certificates_relation = self._tls.get_tls_relation()
+        tls_secret_name = get_config(self, "tls-secret-name")
+
+        if not tls_certificates_relation and not tls_secret_name:
+            event.fail("Certificates relation not created.")
+            return
+
+        if tls_secret_name:
+            event.fail(
+                "Getting certificate from charm configuration is not yet supported",
+            )
+            return
+
+        tls_rel_data = tls_certificates_relation.data[self.app]
+        if tls_rel_data.get(f"certificate-{hostname}"):
+            event.set_results(
+                {
+                    f"certificate-{hostname}": tls_rel_data[f"certificate-{hostname}"],
+                    f"ca-{hostname}": tls_rel_data[f"ca-{hostname}"],
+                    f"chain-{hostname}": tls_rel_data[f"chain-{hostname}"],
                 }
-            },
-        }
+            )
+        else:
+            event.fail("Certificate not available")
+
+    def _on_certificates_relation_created(self, event: RelationCreatedEvent) -> None:
+        """Handle the TLS Certificate relation created event.
+
+        Args:
+            event: The event that fires this method.
+        """
+        if not self._are_relations_ready(event):
+            return
+        hostname = self.get_hostname()
+        if not hostname:
+            self.unit.status = BlockedStatus("Waiting for hostname to be configured")
+            event.defer()
```

Comment:
A config changed will be configured at some point, so you probably don't need to defer here

Reply:
I'm marking this as resolved as per my comment above.

Reply:
Sorry if I've missed this comment - please disregard my above comments on defer if the same applies. (The Github changed files mode didn't quite let me see the prev comments)